src/main/main.ts

```diff
@@ -188,214 +189,195 @@ function initializeApp() {
-            const progress = zipfile.entriesRead / zipfile.entryCount;
-            win?.webContents.send("update extract progress", progress);
-          },
-        });
-        win?.webContents.send("update extract complete");
-        console.log("The zip archive", dlPath, "has extracted to", tempDir);
-        win?.webContents.send("update copying progress");
-        await utils.copyDir(tempDir, extractPath);
-        console.log("Copied extracted files from", tempDir, "to", extractPath);
-        try {
-          await fs.promises.rmdir(tempDir, { recursive: true });
-          console.log("Removed all temporary files from", tempDir);
-        } catch (e) {
-          console.warn(
-            "Failed to remove temporary files from",
-            tempDir,
-            "\n",
-            e
-          );
-        }
-        win?.webContents.send("update copying complete");
-      } else if (process.platform == "darwin") {
-        // .tar.{gz,bz2} 해제
-        const lowerFname = dlFname.toLowerCase();
-        const bz2 =
-          lowerFname.endsWith(".tar.bz2") || lowerFname.endsWith(".tbz");
-        console.log(
-          "Start to extract the tarball archive",
-          dlPath,
-          "to",
-          extractPath
-        );
-        try {
-          await spawnPromise(
-            "tar",
-            [`xvf${bz2 ? "j" : "z"}`, dlPath, "-C", extractPath],
-            { capture: ["stdout", "stderr"] }
-          );
-        } catch (e) {
-          console.error(`${e}:\n`, e.stderr);
-          throw e;
-        }
-        console.log(
-          "The tarball archive",
-          dlPath,
-          "has extracted to ",
-          extractPath
-        );
-      } else {
-        console.warn("Not supported platform.");
-        return;
-      }
+  if (win == null) return;
 
-      // 압축을 푼 뒤 압축 파일은 제거합니다.
-      await fs.promises.unlink(dlPath);
+  console.log("peerVersionExtra (hex):", peerVersionExtra);
+  const buffer = Buffer.from(peerVersionExtra, "hex");
+  console.log("peerVersionExtra (bytes):", buffer);
+  const extra = decode(buffer) as BencodexDict;
+  console.log("peerVersionExtra (decoded):", JSON.stringify(extra)); // 다른 프로세스라 잘 안보여서 JSON으로...
+  const macOSBinaryUrl = extra.get("macOSBinaryUrl") as string;
+  const windowsBinaryUrl = extra.get("WindowsBinaryUrl") as string;
+  console.log("macOSBinaryUrl: ", macOSBinaryUrl);
+  console.log("WindowsBinaryUrl: ", windowsBinaryUrl);
+  const downloadUrl =
+    process.platform === "win32"
+      ? windowsBinaryUrl
+      : process.platform === "darwin"
+      ? macOSBinaryUrl
+      : null;
 
-      // 설정 합치기
-      const newConfig = JSON.parse(
-        await fs.promises.readFile(configPath, { encoding: "utf-8" })
-      );
-      const config = {
-        ...bakConfig,
-        ...newConfig,
-      };
-      await fs.promises.writeFile(configPath, JSON.stringify(config), "utf-8");
+  if (downloadUrl == null) return; // 지원 안 하는 플랫폼이니 무시.
```

Comment:
Can you add some logs here?
```suggestion
  if (downloadUrl == null) {
    console.log(`Stop update process. Not support ${process.platform}.`);
    return;
  }
```